tutorials/javaFxPart3.md

```diff
@@ -7,15 +7,20 @@
 
 Picking up from where we left off last tutorial, we have successfully achieved the desired layout. Now let’s make the application respond to user input.
 
-Rather than to do everything in one try, let’s iterate and build up towards our final goal.
+JavaFX has an _event-driven architecture style_. As such, we programmatically define _handler_ methods to execute as a response to certain _events_. "Events" refers to the [Event class](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html). Two of the most commonly used subclasses are [ActionEvent](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html) (such as a button fired or a key frame finished) and [InputEvent](https://docs.oracle.com/javase/8/javafx/api/javafx/event/Event.html) (such as a mouse click).
+All UI components have a set of methods to define handlers with respect to different events. Some of these methods are unique to the component, but most are inherited from the parent classes `Node`, `Control`, etc.
+For example, the [Button](https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/Button.html) class inherits `setOnAction` from parent `ButtonBase` and `setOnMouseClicked` from `Node`. Either one fulfills the functionality of Duke.
+We pass in a function (the "response") as an argument of these methods to define the behavior of each component.
 
-## Iteration 1 – Echoing the user
+When an event is detected, JavaFX will call the respective handlers.
 
-JavaFX has an _event-driven architecture style_. As such, we programmatically define _handler_ methods to execute as a response to certain _events_. When an event is detected, JavaFX will call the respective handlers.
+Rather than doing everything in one try, let’s iterate and build up towards our final goal.
+
+## Iteration 1 – Echoing the user input
 
 For Duke, there are two events that we want to respond to, namely the user pressing `Enter` in the `TextField` and left-clicking the `Button`. These are the `onAction` event for the `TextField` and the `onMouseClicked` event for the `Button`.
 
-For now, let’s have the application add a new `Label` with the text from the `TextField`. Update the `Main` class as follows. You'll need to add an `import javafx.scene.control.Label;` too.
```

Comment:
Is there a reason for this to be removed? Sorry if I have missed out any context, is this import not necessary?

Reply:
Hi! Sorry I believe I deleted this because of another modification which moves the import for label to part two. I will open new PRs and move this change to the respective PR.